doc/languages-frameworks/cuda.section.md

```diff
@@ -54,3 +54,65 @@ for your specific card(s).
 
 Library maintainers should consult [NVCC Docs](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/)
 and release notes for their software package.
+
+## Adding a new CUDA release {#adding-a-new-cuda-release}
+
+> **WARNING**
+>
+> This section of the docs is still very much in progress. Feedback is welcome in GitHub Issues tagging @NixOS/cuda-maintainers or on [Matrix](https://matrix.to/#/#cuda:nixos.org).
+
+The CUDA Toolkit is a suite of CUDA libraries and software meant to provide a development environment for CUDA-accelerated applications, packaged in a multi-gigabyte monolithic installer. Since CUDA 11.4, NVIDIA has maintained CUDA redistributables (“CUDA-redist”): individually packaged components, meant to facilitate redistribution and inclusion in downstream projects.
+
+All new projects should use the CUDA redistributables, as they are much easier to maintain and update.
+
+### Updating CUDA redistributables {#updating-cuda-redistributables}
+
+1. Go to NVIDIA's index of CUDA redistributables: <https://developer.download.nvidia.com/compute/cuda/redist/>
+2. Copy the `redistrib_*.json` corresponding to the release to `pkgs/development/compilers/cudatoolkit/redist/manifests`.
+3. Generate the `redistrib_features_*.json` file by running:
+
+    ```bash
+    nix run github:ConnorBaker/cuda-redist-find-features -- <path to manifest>
+    ```
+
+    That command will generate the `redistrib_features_*.json` file in the same directory as the manifest.
+
+4. Include the path to the new manifest in `pkgs/development/compilers/cudatoolkit/redist/extension.nix`.
+
+### Updating the CUDA Toolkit {#updating-the-cuda-toolkit}
+
+> **WARNING**
+>
+> While the CUDA Toolkit is still available in Nixpkgs, it is not recommended for use and should be considered deprecated.
```

Comment:
This portion e.g. _I thought I had commented on the PR earlier, but I now suspect must've left the comment at the commit or something (is github getting more confusing over time?)_

Back to the point, the phrase "CUDA Toolkit ... should be considered deprecated" probably conveys the wrong idea. The redist-n-splayed distribution of the cuda toolkit is still cuda toolkit. What's being deprecated is the `cudaPackages.cudatoolkit` attribute, associated with the runfile-based cuda distribution. I think it's easiest that we directly refer to the attributes when we talk about these things in the documentation

Reply:
> This portion e.g. _I thought I had commented on the PR earlier, but I now suspect must've left the comment at the commit or something (is github getting more confusing over time?)_

A force-push would have nuked that wouldn't it 😬

> Back to the point, the phrase "CUDA Toolkit ... should be considered deprecated" probably conveys the wrong idea. The redist-n-splayed distribution of the cuda toolkit is still cuda toolkit. What's being deprecated is the `cudaPackages.cudatoolkit` attribute, associated with the runfile-based cuda distribution. I think it's easiest that we directly refer to the attributes when we talk about these things in the documentation

Ah, I didn't realize the redistributables were considered the CUDA Toolkit. I'll update the language in the docs to reflect that!

I thought they were something else entirely: individually packaged and usable CUDA libraries.

Reply:
> I thought they were something else entirely: individually packaged and usable CUDA libraries

Idk, it's a mess, let's just mimic nvidia when we have to use some names, and let's also keep the number of names to a minimum xD

- https://docs.nvidia.com/cuda/cuda-installation-guide-linux/#parsing-redistrib-json
- The link named "CUDA Toolkit" from the table referenced above leads to: https://developer.download.nvidia.com/compute/cuda/redist/

Reply:
Fixed in the latest